### PR TITLE
Derive new addresses (part 3)

### PR DESCRIPTION
**Pull request by nop33**
- Installing Electron React dev tools (the extension needs to already be installed in Chrome/Brave).
   image
- Fetching and displaying confirmed transactions (pending transactions PR coming up)

   image
   image

### CODE REVIEW

File: src/components/Badge.tsx [hunk 1]
```diff
@@ -33,40 +31,13 @@ interface BadgeProps {
...
-      color = theme.name === 'dark' ? 'rgba(255, 255, 255, 0.8)' : 'rgba(255, 255, 255, 1)'
-      break
-    case 'neutralHighlight':
-      backgroundColor = 'rgba(101, 16, 247, 1)'
-      color = 'rgba(255, 255, 255, 1)'
-  }
-
-  return { backgroundColor, color }
-}
-
```
Comment (nop33) [inline src/components/Badge.tsx:64]: I moved all the color definitions in the theme, let me know if you disagree

File: src/pages/Wallet/AddressDetailsPage.tsx [hunk 2]
```diff
@@ -126,6 +174,31 @@ const AddressDetailsPage = () => {
   )
 }
 
+const IOList = ({ currentAddress, isOut, outputs, inputs, timestamp }: IOListProps) => {
```
Comment (nop33) [inline src/pages/Wallet/AddressDetailsPage.tsx:177]: I am reusing the component of the wallet home page, which will eventually be adjusted, and this will move to a new component to keep the code dry.

File: src/style/styled.d.ts [hunk 3]
```diff
@@ -49,5 +49,19 @@ declare module 'styled-components' {
       valid: string
       highlightGradient: string
     }
+    badge: {
+      font: {
+        plus: string
```
Comment (nop33) [inline src/style/styled.d.ts:54]: I did not use the bright new colors from Figma, because personally, I find the "pastel'-er version of the green and red more pleasing. However, I trust your opinion more! We can easily change the colors in the theme settings.

Comment (mvaivre) [reply src/style/styled.d.ts:54]: Fine by me, I was still iterating. I'll do a final tweaking pass at the very end 🎨

**Comment by mvaivre** [comment]
I forgot a little thing in the previous PR: it would be great to show the text cursor (blinker) in the `ColoredLabelInput`. 
Tricky, but feasible I believe :) Do you prefer to take a look at this in this PR, or do you prefer to integrate this in the next one?

**Comment by nop33** [comment]
Nice catch. I'll try to figure it out on this one.

**Comment by nop33** [comment]
@mvaivre done!

**Comment by nop33** [comment]
https://user-images.githubusercontent.com/1579899/152171369-e5aa9b73-7673-4b9b-a490-89351e202981.mp4

**Comment by mvaivre** [comment]
> Kapture.2022-02-02.at.15.18.23.mp4

Credible!!! <3

File: src/components/Badge.tsx [hunk 4]
```diff
@@ -33,40 +31,13 @@ interface BadgeProps {
...
 Badge = styled(Badge)`
-  background-color: ${({ type, theme }) => getBadgeColor(type, theme).backgroundColor};
-  color: ${({ type, theme }) => getBadgeColor(type, theme).color};
+  color: ${({ type, theme }) => theme.badge.font[type]};
```
Comment (mvaivre) [inline src/components/Badge.tsx:39]: A badge without a background color isn't really a badge anymore I would say, but just a stylised string.
Your `DarkLabel` (used for from/to labels) looks more like a badge.

That's a purely style related discussion we should have. In the explorer, I'm using badges for amounts in tables + many other things. In the wallet, I tried another approach to make the UI feel less cluttered (we will have more content to display than in the explorer). Which approach do you prefer?

IMO I tend to think that badges should be used carefully for things we wanna highlight or label, be it for address labels, important status, total amounts etc.. Badges can be with or without stroke, depending on the visual importance we wanna give.

We would keep repeating amounts (in table) or other inline data as simple strings (and make the change in the explorer as well).

As a side note, I'd like to reduce a bit the use of colors accros my mockups (for instance, not use green and red for every amount diff in a table). I'll experiment around that.

**TLDR: Let's keep the badge component, but let's not use it for TX "direction" label or table amounts. :) And I need to keep iterating.**

File: src/pages/Wallet/AddressDetailsPage.tsx [hunk 5]
```diff
@@ -162,4 +235,15 @@ const MainAddress = styled.div`
   position: absolute;
 `
 
+const DarkLabel = styled.span`
```
Comment (mvaivre) [inline src/pages/Wallet/AddressDetailsPage.tsx:238]: This could actually use the `Badge` component.

File: src/components/Badge.tsx (hunk 4 above)
Comment (mvaivre) [reply src/components/Badge.tsx:39]: I'm making changes to the mockups right now, so don't spend too much time on this. This was mostly to keep you up to date with my thoughts 🧠

File: src/contexts/addresses.tsx [hunk 6]
```diff
@@ -34,13 +34,17 @@ import { useGlobalContext } from './global'
...
   publicKey: string
   privateKey: string
   group: number
   index: number
   settings: AddressSettings
   details?: AddressInfo
+  transactions?: {
+    confirmed: Transaction[]
+  }
+  lastUsed?: number
```
[the remark this answers is not in the document]
Comment (nop33) [reply src/contexts/addresses.tsx:47]: yes, makes sense!

File: src/components/Badge.tsx (hunk 4 above)
Comment (nop33) [reply src/components/Badge.tsx:39]: Thanks for sharing your thoughts. I agree with all of them.

File: src/pages/Wallet/AddressDetailsPage.tsx (hunk 5 above)
Comment (nop33) [reply src/pages/Wallet/AddressDetailsPage.tsx]: Updated. I am sure we'll update it again once we re-iterate over the styles of labels and badges.

**Comment by nop33** [comment]
@mvaivre, All comments were addressed! 🙏

**Comment by mvaivre** [comment]
LGTM then 🚀
